### PR TITLE
Support for special character in pdf password

### PR DESCRIPTION
**Pull request by rakshith49**
Added Support for special character in pdf password validation, previously crashing for special character (java.lang.ArrayIndexOutOfBoundsException)

### CODE REVIEW

File: src/com/sun/pdfview/PDFDocCharsetEncoder.java [hunk 1]
```diff
@@ -52,12 +52,13 @@
...
     static
     {
-        for (byte i = 0; i < PDFStringUtil.PDF_DOC_ENCODING_MAP.length; ++i) {
+       for (int i = 0; i < PDFStringUtil.PDF_DOC_ENCODING_MAP.length; i++) {
             final char c = PDFStringUtil.PDF_DOC_ENCODING_MAP[i];
             final boolean identical = (c == i);
             IDENT_PDF_DOC_ENCODING_MAP[i] = identical;
             if (!identical) {
-                EXTENDED_TO_PDF_DOC_ENCODING_MAP.put(c, i);
+                byte value = (byte) PDFStringUtil.PDF_DOC_ENCODING_MAP[i];//support for special character in pdf password
```
Comment (saket) [inline src/com/sun/pdfview/PDFDocCharsetEncoder.java:60]: Can you add some space before the comment and make the first letter capital?

**Comment by saket** [comment]
I was not expecting anyone to use this library :)

Three things:
1. Can you rebase your changes with `master` again?
2. The changed code does not belong to me and I have no idea how it works. So I am unsure how to go about this.
3. Do you have a source for the solution?

**Comment by rakshith49** [comment]
Actually, now im working in cleartax, we are using same library it had
issue, so resolved that one and updated to you...

**Comment by saket** [comment]
Oh okay. Where did you find the solution for this? I'm asking because your solution involves putting the same value as the key and as the value of the map, which looks suspicious.

**Comment by rakshith49** [comment]
I only added one line of code... Through debugging & tested it for every
scenarios

**Comment by saket** [comment]
So you.. found the solution by trial and error, on your own?

**Comment by rakshith49** [comment]
Not trail and error, debugged it properly and added that line of code. Any
way library was crashing, handled proper way...  I can provide well explanations about that code

**Comment by saket** [comment]
Okay. Then I recommend you submit this PR instead to the [original library here](https://github.com/katjas/PDFrenderer) and I'll pull in the changes. It'll be better if they review your solution since I did not write that code.

**Pull request closed without merging**